build-system/tasks/dep-check.js

```diff
@@ -178,7 +178,11 @@ async function getEntryPointModule() {
     .map((x) => `extensions/${x}`)
     .filter((x) => fs.statSync(x).isDirectory())
     .map(getEntryPoint);
-  const allEntryPoints = flatten(extensionEntryPoints).concat(coreBinaries);
+  const vendors = await fs.promises.readdir('3p/vendors');
```

Comment:
If you're going to do `await fs.promies.readdir` why not just call `fs.readdirSync` directly? :D

Reply:
My impression was that the `await` version was always preferred otherwise you block any other action that could happen concurrently. For example processing multiple `getEntryPointModules` concurrently would be sped up by using the async vs sync call